tests/providers/slack/chat-message.ts

```diff
@@ -0,0 +1,195 @@
+let provider: BaseProvider;
+let handlerName = "chat-message";
+let testConfig: GenericTestConfig;
+
+// Configure provider and handler without certain attributes
+let providerConfig: Omit<BaseProviderConfig, "sbtImage" | "label"> = {};
+let handlerConfig: SlackHandlerConfig = {
+  messagesPerGroupLimit: 3
+};
+
+// Test suite for Slack Chat Message syncing
+describe(`${providerId} chat message tests`, function () {
+  this.timeout(100000);
+
+  // Before all tests, set up the network, connection, and provider
+  this.beforeAll(async function () {
+    network = await CommonUtils.getNetwork();
+    connection = await CommonUtils.getConnection(providerId);
+    provider = Providers(providerId, network.context, connection);
+
+    // Configure test settings
+    testConfig = {
+      idPrefix: `${provider.getProviderId()}-${connection.profile.id}`,
+      timeOrderAttribute: "insertedAt",
+      batchSizeLimitAttribute: "batchSize",
+    };
+  });
+
+  // Test fetching data for Slack Chat
+  describe(`Fetch ${providerId} data`, () => {
+
+    it(`Can pass basic tests: ${handlerName}`, async () => {
+      // Build the necessary test objects
+      const { api, handler, provider } = await CommonTests.buildTestObjects(
+        providerId,
+        SlackChatMessageHandler,
+        providerConfig,
+        connection
+      );
+
+      // Set the handler configuration
+      handler.setConfig(handlerConfig);
+
+      try {
+        // Set up initial sync position
+        const syncPosition: SyncHandlerPosition = {
+          _id: `${providerId}-${handlerName}`,
+          providerId,
+          handlerId: handler.getId(),
+          accountId: provider.getAccountId(),
+          status: SyncHandlerStatus.ENABLED,
+        };
+
+        // Start the sync process
+        const response = await handler._sync(api, syncPosition);
+        const results = <SchemaRecord[]>response.results;
+
+        // Extract chat groups and messages from the results
+        const chatGroups = <SchemaSocialChatGroup[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_GROUP);
+        const chatMessages = <SchemaSocialChatMessage[]>results.filter(result => result.schema === CONFIG.verida.schemas.CHAT_MESSAGE);
+
+        // Ensure results are returned
+        assert.ok(results && results.length, "Have results returned");
+
+        // Check IDs in the returned items
+        CommonTests.checkItem(results[0], handler, provider);
+
+        // Verify sync status is active
+        assert.equal(
+          SyncHandlerStatus.SYNCING,
+          response.position.status,
+          "Sync is active"
+        );
+
+        // Ensure the message batch per group works
+        const firstGroupId = chatMessages[0].groupId;
+        const firstGroupMessages = chatMessages.filter(msg => msg.groupId === firstGroupId);
+        assert.equal(firstGroupMessages.length, handlerConfig.messagesPerGroupLimit, "Processed correct number of messages per group");
+
```

Comment:
Add a test to make sure most recent messages are processed first

Reply:
> Add a test to make sure most recent messages are processed first

Done